ApollosProject/apollos-prototype
# Algolia Delta Sync

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds a delta sync queue/job in order to reduce the weight of our Algolia syncs. This will allow us to sync only items that have been created/modified after a certain date. This should drastically reduce our Algolia syncing.

Previously we did a full-sync every day which also included removing every record first. That job has been moved to once a week. The new delta sync will run daily and should have a comparatively minimal amount of records to update.

### What design trade-offs/decisions were made?

I added a new cursor to the ContentItem data source. It functions similarly to byActive, but only fetches after a given date. The date is parsed from a queue that gets passed in, so it can't actually be used in any other situation.

I'm thinking that maybe I should do the date parsing earlier and then just pass the date down instead of the entire queue.

### How do I test this PR?

1. Perform a full sync by uncommenting ln 40 of `/apollos-data-connector-algolia/src/jobs.js`
2. Create a new piece of published content in ROCK
3. Perform a Delta Sync by recommenting ln 40 and uncommenting ln 41 of `/apollos-data-connector-algolia/src/jobs.js` ; the new piece of content should be searchable
5. Update the content item (I like to change the title)
6. Perform another delta sync; the content should be searchable and updated

### What automated tests did you write?

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
